File: Frontend.Angular/src/app/models/lesson-filter.ts
export interface LessonFilter {
    recipientName?: string;
    topic?: string;
    dateRange?: Date[];
    startDate?: string;
    endDate?:string;
    minDuration? :number;
    maxDuration? :number;
    minPrice?: number;
    maxPrice?: number;
    status?: number;
}

File: api/Avancira.Application/Lessons/Dtos/LessonFilter.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace Avancira.Application.Lessons.Dtos
{
    public class LessonFilter
    {
        public int Page { get; set; } = 1;
        public int PageSize { get; set; } = 10;
        public string? RecipientName { get; set; }
        public string? Topic { get; set; }

        public DateTime? StartDate { get; set; }
        public DateTime? EndDate { get; set; }

        public int? MinDuration { get; set; }
        public int? MaxDuration { get; set; }

        public decimal? MinPrice { get; set; }
        public decimal? MaxPrice { get; set; }

        public int Status { get; set; } = -1;
    }
}
